suite-common/wallet-core/src/fiat-rates/fiatRatesReducer.ts

```diff
@@ -74,6 +76,15 @@ export const prepareFiatRatesReducer = createReducerWithExtraDeps(
                     isLoading: false,
                     error: action.error.message || `Failed to update ${ticker.symbol} fiat rate.`,
                 };
+            })
+            .addCase(fiatRatesActions.addFiatRatesForTimestamps, (state, action) => {
+                const { ticker, localCurrency, rates } = action.payload;
+                const fiatRateKey = getFiatRateKeyFromTicker(ticker, localCurrency);
+
+                state['historic'][fiatRateKey] = {
+                    ...state['historic'][fiatRateKey],
+                    ...rates.reduce((acc, rate) => ({ ...acc, [rate.timestamp]: rate.rate }), {}),
```

Comment:
I have little bit troubles to understand what is reduce doing here, can you elaborate little bit?

Reply:
I think that with readability it's better how it's done now. Maybe a comment will help for future readers what this actually does.